starfish/pipeline/filter/_base.py

```diff
@@ -0,0 +1,31 @@
+# todo can we abstract this a single time? It feels unnecessary to have multiple definitions of this class.
+class FilterAlgorithmBase(object):
+    @classmethod
+    def from_cli_args(cls, args):
+        """
+        Given parsed arguments, construct an instance of this filter algorithm.
+
+        Generally, this involves retrieving the appropriate attributes from args to pass to the filter algorithm's
+        constructor.
+        """
+        raise NotImplementedError()
+
+    @classmethod
+    def get_algorithm_name(cls):
+        """
+        Returns the name of the algorithm.  This should be a valid python identifier, i.e.,
+        https://docs.python.org/3/reference/lexical_analysis.html#identifiers
+        """
+        raise NotImplementedError()
+
+    @classmethod
+    def add_arguments(cls, group_parser):
+        """Adds the arguments for the algorithm."""
+        raise NotImplementedError()
+
+    # @ttung classmethod means we can call this from the API without calling __init__
```

Comment:
It appears to pass the tests despite this method taking parameters, unless I'm misunderstanding.

Reply:
Well, you don't call `FilterAlgorithmBase.filter`, right...?

Reply:
Well, this can _inherit_ that method and call it. However, given our conversation, and the `instance(parameters).run(stack)` pattern, we'll never call a classmethod. So, this is moot.